exercises/practice/simple-cipher/src/test/java/SimpleCipherTest.java

```diff
@@ -1,113 +1,104 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-
-@RunWith(Enclosed.class)
 public class SimpleCipherTest {
-    public static class RandomKeyCipher {
-        private Cipher cipherWithDefaultKey;
-
-        @Before
-        public void setup() {
-            cipherWithDefaultKey = new Cipher();
-        }
+    private Cipher randomKeyCipher;
+    private Cipher substitutionCipher = new Cipher("abcdefghij");
 
-        /**
-         * Here we take advantage of the fact that plaintext of "aaa..." doesn't output the key. This is a critical
-         * problem with shift ciphers, some characters will always output the key verbatim.
-         */
-        @Test
-        public void cipherCanEncode() {
-            String plainText = "aaaaaaaaaa";
-            String cipherText = cipherWithDefaultKey.getKey().substring(0, 10);
-            assertThat(cipherWithDefaultKey.encode(plainText)).isEqualTo(cipherText);
-        }
+    @Before
+    public void setup() {
+        randomKeyCipher = new Cipher();
+    }
 
-        @Ignore("Remove to run test")
-        @Test
-        public void cipherCanDecode() {
-            String cipherText = "aaaaaaaaaa";
-            assertThat(cipherWithDefaultKey.decode(cipherWithDefaultKey.getKey().substring(0, 10)))
-                .isEqualTo(cipherText);
-        }
+    /**
+     * Here we take advantage of the fact that plaintext of "aaa..." doesn't output the key. This is a critical
+     * problem with shift ciphers, some characters will always output the key verbatim.
+     */
+    @Test
+    public void randomKeyCipherCanEncode() {
+        String plainText = "aaaaaaaaaa";
+        String cipherText = randomKeyCipher.getKey().substring(0, 10);
+        assertThat(randomKeyCipher.encode(plainText)).isEqualTo(cipherText);
+    }
 
-        @Ignore("Remove to run test")
-        @Test
-        public void cipherIsReversible() {
-            String plainText = "abcdefghij";
-            assertThat(cipherWithDefaultKey.decode(cipherWithDefaultKey.encode(plainText))).isEqualTo(plainText);
-        }
+    @Ignore("Remove to run test")
+    @Test
+    public void randomKeyCipherCanDecode() {
+        String cipherText = "aaaaaaaaaa";
+        assertThat(randomKeyCipher.decode(randomKeyCipher.getKey().substring(0, 10)))
+            .isEqualTo(cipherText);
+    }
 
-        @Ignore("Remove to run test")
-        @Test
-        public void keyIsLowercaseLetters() {
-            assertThat(cipherWithDefaultKey.getKey()).matches("^[a-z]+$");
-        }
+    @Ignore("Remove to run test")
+    @Test
+    public void randomKeyCipherIsReversible() {
+        String plainText = "abcdefghij";
+        assertThat(randomKeyCipher.decode(randomKeyCipher.encode(plainText))).isEqualTo(plainText);
     }
 
-    public static class SubstitutionCipher {
-        private Cipher cipherWithDefaultKey = new Cipher("abcdefghij");
+    @Ignore("Remove to run test")
+    @Test
+    public void keyIsLowercaseLetters() {
```

Comment:
I think it makes sense to rename this to `randomCipherKeyIsLowercaseLetters()` as well, since the test only makes sense in the context of a randomly generated key.

Reply:
You're right, I should have renamed that too. I renamed this method to `randomKeyCipherKeyIsLowercaseLetters` for consistency.